lib/session_encryptor.rb

```diff
@@ -39,10 +40,17 @@ class SensitiveValueError < StandardError; end
   def load(value)
     return LegacySessionEncryptor.new.load(value) if should_use_legacy_encryptor_for_read?(value)
 
-    _v2, ciphertext = value.split(':')
+    payload = MessagePack.unpack(value)
+    ciphertext = payload['t']
+    compressed = payload['c']
     decrypted = outer_decrypt(ciphertext)
+    decrypted = if compressed == 1
+                  Zlib.gunzip(decrypted)
+    else
+      decrypted
+    end
```

Comment:
I feel like rubocop is not doing us any favors here... I'd expect:
```suggestion
    decrypted = if compressed == 1
                  Zlib.gunzip(decrypted)
                else
                  decrypted
                end
```